spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,290 @@
+
+    Parameters
+    ----------
+    traces : 2D np.array
+        (num_channels x num_samples) numpy array with the signals
+    n_channel_pad : int
+        Number of channels to pad
+    n_channel_taper : int
+        Number of channels to taper
+    agc_options : dict
+        Automatic Gain Control options
+    sos_filter : sos filter
+        Butterworth filter in second-order sections format
+
+    Returns
+    -------
+    traces_kfilt : 2D np.array
+        The spatially filtered traces
+    """
+    num_channels = traces.shape[1]
+
+    # lateral padding left and right
+    num_channels_padded = num_channels + n_channel_pad * 2
+
+    # apply agc and keep the gain in handy
+    if not agc_options:
+        gain = 1
+    else:
+        traces, gain = agc(traces,
+                           window_length=agc_options["window_length_s"],
+                           sampling_interval=agc_options["sampling_interval"])
+
+    if n_channel_pad > 0:
+        # pad the array with a mirrored version of itself and apply a cosine taper
+        traces = np.c_[np.fliplr(traces[:, :n_channel_pad]),
+                       traces,
+                       np.fliplr(traces[:, -n_channel_pad:])]
+
+    if n_channel_taper > 0:
+        taper = fcn_cosine([0, n_channel_taper])(np.arange(num_channels_padded))  # taper up
+        taper *= 1 - fcn_cosine([num_channels_padded - n_channel_taper, num_channels_padded])(np.arange(num_channels_padded))   # taper down
+        traces = traces * taper[np.newaxis, :]
+
+    traces = scipy.signal.sosfiltfilt(sos_filter, traces, axis=1)
+
+    if n_channel_pad > 0:
+        traces = traces[:, n_channel_pad:-n_channel_pad]
+
+    return traces * gain
+
+# -----------------------------------------------------------------------------------------------
+# IBL Helper Functions
+# -----------------------------------------------------------------------------------------------
+
+def agc(traces, window_length, sampling_interval, epsilon=1e-8):
+    """
+    Automatic gain control
+    w_agc, gain = agc(w, window_length=.5, si=.002, epsilon=1e-8)
+    such as w_agc * gain = w
+    :param traces: seismic array (sample last dimension)
+    :param window_length: window length (secs) (original default 0.5)
+    :param si: sampling interval (secs) (original default 0.002)
+    :param epsilon: whitening (useful mainly for synthetic data)
+    :return: AGC data array, gain applied to data
+    """
+    num_samples_window = int(np.round(window_length / sampling_interval / 2) * 2 + 1)
+    window = np.hanning(num_samples_window)
+    window /= np.sum(window)
+
+    gain = scipy.signal.fftconvolve(np.abs(traces), window[:, None], mode='same', axes=0)
+
+    gain += (np.sum(gain, axis=0) * epsilon / traces.shape[0])[np.newaxis, :]
+
+    dead_channels = np.sum(gain, axis=0) == 0
+
+    traces[:, ~dead_channels] = traces[:, ~dead_channels] / gain[:, ~dead_channels]
+
+    return traces, gain
```

Comment:
I have the feeling that this gain control will be computed in every chunk.
And so the results will depend on the chunk_size itself.
I am not squre this is a good idea.
Maybe we should compute this once at the begenning no ?